fs/walk_test.ts

```diff
@@ -0,0 +1,202 @@
+import { cwd, chdir, FileInfo, makeTempDir, mkdir, open, remove, symlink } from "deno";
+
+import { walk } from "./walk.ts";
+import { test, assert, TestFunction } from "../testing/mod.ts"; 
+
+async function testWalk(setup: (string) => void | Promise<void>, t: TestFunction): Promise<void> {
+  const name = t.name;
+  async function fn() {
+    const orig_cwd = cwd();
+    const d = await makeTempDir();
+    chdir(d);
+    try {
+     await setup(d)
+     await t()
+    } finally {
+      chdir(orig_cwd);
+      remove(d, {recursive: true});
+    }
+  }
+  test({ name, fn });
```

Comment:
I like this pattern of wrapping the test function for setup.

Reply:
Bert suggested this in discussion of setup/teardown for testFileserver - I should push a branch with that change too...